analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/ExecutingSqlQueries.cs

```diff
@@ -81,10 +82,56 @@ public ExecutingSqlQueries()
             && expression is BinaryExpressionSyntax concatenation
             && !IsConcatenationOfConstants(concatenation, semanticModel);
 
-        private bool IsTrackedVariableDeclaration(ExpressionSyntax argument, SemanticModel semanticModel) =>
-            (argument is IdentifierNameSyntax identifierNameSyntax
-             && semanticModel.GetDeclaringSyntaxNode(identifierNameSyntax) is VariableDeclaratorSyntax variableDeclaratorSyntax
-             && IsTracked(variableDeclaratorSyntax.Initializer?.Value, semanticModel));
+        private static bool IsTrackedVariableDeclaration(ExpressionSyntax expression, BaseContext context)
+        {
+            if (expression is IdentifierNameSyntax)
+            {
+                var node = expression;
+                Location location;
+                do
+                {
```

Comment:
This should make your life easier with number of conditions and you won't need the `(node as IdentifierNameSyntax)` later
```suggestion
            var node = expression;
            Location location;
            while (node is IdentifierNameSyntax is identifierName)
            {
```

Reply:
The key here was to remove the redundant `if (expression is IdentifierNameSyntax)` check